### PR TITLE
Updated flavors to include m5 instances

### PR DESCRIPTION
**Pull request by rh-dluong**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1555046

### CODE REVIEW

**Comment by robbmanes** [comment]
Looked at this with @rh-dluong and we can't figure out where the style error is at.  When I just `rubocop` the file itself I get about 2k errors anyway, and I can't find any specific `rubocop` config to map it too.

He is also fixing the spec [here](https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/spec/models/manageiq/providers/amazon/aws_refresher_spec_counts.rb#L148) to accommodate the increased count in flavors.

**Comment by blomquisg** [comment]
Overall this looks good.  I haven't validated the specific details of each flavor being added.  @robbmanes do you mind doing that?  Just to give us the peace of mind that it's adding all the right numbers, etc...

There's another approach that we're looking at, which is to backport https://github.com/ManageIQ/manageiq-providers-amazon/pull/289.

But, I don't see these two approaches as mutually exclusive.  We can both update the core instance_types file as well as backport functionality to allow users to manage their flavors as new flavors are released.

**Comment by robbmanes** [comment]
Sure, I'll check the flavor specifics.  I also agree they're not mutually exclusive, and having both `instance_types` and custom flavors would be nice.

**Comment by robbmanes** [comment]
Confirmed instance sizing and attributes via [Amazon's Instance Page](https://aws.amazon.com/ec2/instance-types/) and my own EC2 account by spinning them up and running `aws ec2 describe-instance-attribute` on them. LGTM sans the robocop style errors.

File: app/models/manageiq/providers/amazon/instance_types.rb [hunk 1]
```diff
@@ -342,6 +342,150 @@ module ManageIQ::Providers::Amazon::InstanceTypes
       :vpc_only                => true,
     },
 
+    "m5.large"    => {
```
Comment (bronaghs) [inline app/models/manageiq/providers/amazon/instance_types.rb]: @rh-dluong - Try moving the => over to the right to align with the hash rockets inside the hash

**Comment by jameswnl** [comment]
@rh-dluong Just want to make sure that the update to this `instance_types.rb` you have is generated/updated by running the `rake` task `manageiq-providers-amazon/lib/tasks_private/instance_types.rake`. 

You should be able run the upstream version of the task (i.e. non-Fine version is fine) and submit the new `instance_types.rb`

File: app/models/manageiq/providers/amazon/instance_types.rb (hunk 1 above)
Comment (rh-dluong) [reply app/models/manageiq/providers/amazon/instance_types.rb]: Hm, that didn't seem to work either.  Thanks for the suggestion though!

**Comment by rh-dluong** [comment]
@jameswnl I did not, but I just rebased using that and sent it up using CFME 5.9

**Comment by bronaghs** [comment]
Thanks for this PR @rh-dluong.
Someone on the Amazon provider team can address the rubocop error in a separate PR.

**Comment by bronaghs** [comment]
@jameswnl - any other questions or are we good to merge?

**Comment by jameswnl** [comment]
@rh-dluong run the rake task to update the `instance_types.rb`

inside `manageiq-providers-amazon` folder
```
curl https://raw.githubusercontent.com/powdahound/ec2instances.info/master/www/instances.json > lib/tasks_private/instance_types_data/instances.json
bundle exec rake instance_types
mv instance_types.rb app/models/manageiq/providers/amazon/instance_types.rb
```

**Comment by rh-dluong** [comment]
@jameswnl I have done as you've asked and pushed the commit.  Let me know if you need anything else.

**Comment by jameswnl** [comment]
@rh-dluong looks like it added a bunch new m5.* instance types. If you can find what you set out to add, then it's LGTM.

**Comment by rh-dluong** [comment]
@jameswnl I had to change one line since it was failing the build.  It was this line:

:vcpu                    => N/A,

I changed it to 72.  I'm assuming it generated N/A since it's a  bare metal instance.  It's also in preview.  Let me know if you want to include this or drop it.  I'm assuming it'll be a while before this PR actually gets released, so it could be out of preview by then.  

Got the info from here:
https://aws.amazon.com/ec2/instance-types/i3/

**Comment by jameswnl** [comment]
@rh-dluong good find. This is definitely a new one.

@bronaghs please decide if the rake task should be fixed for this case now, or it can wait til Alex's new way.  In any case, I'm ok that @rh-dluong manually patch the `instance_types.rb` this time (we won't miss this because it will be caught again next time)

**Comment by bronaghs** [comment]
@rh-dluong - Where did you have to make that change? Can you paste in a link to the line?

File: app/models/manageiq/providers/amazon/instance_types.rb [hunk 2]
```diff
@@ -1421,73 +1085,745 @@ module ManageIQ::Providers::Amazon::InstanceTypes
...
+      :cluster_networking      => nil,
+      :vpc_only                => true,
+    },
+
+    "i3.metal"    => {
+      :name                    => "i3.metal",
+      :family                  => "Unknown",
+      :description             => "I3 High I/O Metal",
+      :memory                  => 512.0.gigabytes,
+      :vcpu                    => 72,
```
Comment (rh-dluong) [inline app/models/manageiq/providers/amazon/instance_types.rb]: This is where I had to change it @bronaghs

**Comment by bronaghs** [comment]
Let's get this PR in. Alex is revamping the rake task anyway in the near term.
As soon as the test pass I will merge.

**Comment by bronaghs** [comment]
@rh-dluong - can you pull from upstream/ master and push? My tests are passing locally.

**Comment by rh-dluong** [comment]
@bronaghs Have done as requested and it seems to have passed.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 4cbbea043c23621fbe8a63e9ee33e6b906b97da2
Author: Bronagh Sorota <bsorota@redhat.com>
Date:   Fri Mar 16 13:55:44 2018 -0400

    Merge pull request #420 from rh-dluong/update-amazon-flavors
    
    Updated flavors to include m5 instances
    (cherry picked from commit 3a4d75533a81deb91ebbcdc54cfc05beb388e5a1)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1558038
```

**Comment by simaishi** [comment]
@rh-dluong @bronaghs There are good amount of conflict getting this PR to Fine branch. Can you please create a separate PR for Fine?

**Comment by rh-dluong** [comment]
Hey @simaishi , I'd do it but my dev env is broken due to gems.  Currently debugging.

**Comment by rh-dluong** [comment]
@simaishi I have created a pull request on the fine branch here: https://github.com/ManageIQ/manageiq-providers-amazon/pull/429

**Comment by simaishi** [comment]
Thank you @rh-dluong !

**Comment by simaishi** [comment]
Backported to Fine via https://github.com/ManageIQ/manageiq-providers-amazon/pull/429

**Pull request merged**